DragonLoop/DragonLoopApp/DragonLoopApp/Views/MapPage.xaml.cs

```diff
@@ -29,5 +32,41 @@ protected override void OnAppearing()
             if (viewModel.Routes == null || !viewModel.Routes.Any())
                 viewModel.LoadRoutesCommand.Execute(null);
         }
+
+        private async void Handle_Route_Toggle(object sender, Xamarin.Forms.ToggledEventArgs e)
+        {
+            //set up stop on screen
+            if (e.Value == true)
+            {
+                ViewCell cell = (sender as Switch).Parent.Parent as ViewCell;
+                Route route = cell.BindingContext as Route;
+
+                if (route != null)
+                {
+                    await viewModel.LoadRouteStops(route.RouteId);
+                    
+                    //Load pins onto map
+                    List<Stop> stops = new List<Stop>(viewModel.RouteStops);
+                    for(int i = 0; i < viewModel.RouteStops.Count(); i++)
+                    {
+                        var pin = new Pin()
+                        {
+                            Position = new Position(Decimal.ToDouble(stops[i].XCoordinate), Decimal.ToDouble(stops[i].YCoordinate)),
+                            Label = stops[i].Name
+                        };
+                        viewModel.Map.Pins.Add(pin);
+                    }
+
+                    //TODO: Create a route -> This requires a custom map object:
+                    // https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/custom-renderer/map/polyline-map-overlay
+                }
+            }
+            //clear stops on screen
+            else
+            {
+                viewModel.Map.Pins.Clear();
```

Comment:
This is what you brought up in your demo I believe, but this will need to just clear the pins for the specific route. You can probably get the Route context at the top level of this method and then branch on the if (e.Value) and either add or remove the pins.

Reply:
So I actually did some brain storming on this...I wanted to add a property on each pin that had the RouteId attached to it and remove it based on that?

Reply:
```suggestion
               //Remove pins from stuff disabled
                if(route != null)
                {
                    await viewModel.LoadRouteStops(route.RouteId);

                    List<Pin> stopsRemove = viewModel.Map.Pins.Where(s => s.Address == route.RouteId.ToString()).ToList();

                    if (stopsRemove.Count > 0)
                    {
                        foreach(var pin in stopsRemove)
                        {
                            viewModel.Map.Pins.Remove(pin);
                        }
                    }
                }
```

Reply:
There's some minor stuff you can clean up here, but I think this approach could work. You probably wouldn't have to load stops here though right? They should already be loaded if you're trying to remove them. What if you made RouteStops a Dictionary<int, IEnumerable<Stop>> where the key is the RouteId and then it is a list of stops? Then you could remove based of the RouteId. Thoughts?

Reply:
The problem is that I am using the address property but that property shows on the actual pin

Reply:
If there is no way to have any hidden metadata on the Pin that might be the way to go. It would also allow us to use images as Pins in the future which I think we'd like to do so we'd probably have to do this anyway.

One more thought I had is that using a Dictionary<int, IEnumerable<Stop>> where the key is the RouteId and then it is a list of stops might be the way to go for this anyway. With the current implementation, when you're removing pins from the map, they are remaining in your IEnumerable which could cause issues when you go to select another route later on.

I'm thinking you could load all of the stops for all routes when the page loads and fill up the dictionary I mentioned above. Then your slider method would just have to either add a pin for each Stop in the IEnumerable for the given route, and do the same clearing based on Route property you're talking about.

Reply:
I guess I could do that and it would limit the number of calls to the database. If the slider is true then it would check the the dictionary and add it to the Pin list in the map using Linq...I think we should wait on this implementation maybe until we do the Custom Map?